imap_processing/run_processing.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+"""Serves as a command line utility to run the processing for a specific
+instrument and data level.
+
+Use
+---
+    python run_processing.py <instrument> <data_level>
+"""
+
+import argparse
+import sys
+from abc import ABC, abstractmethod
+
+from imap_processing import instruments, processing_levels
+
+
+def _parse_args():
+    """Parse the command line arguments.
+
+    Returns
+    -------
+    args : argparse.Namespace
+        An object containing the parsed arguments and their values
+    """
+    description = (
+        "This command line program invokes the processing pipeline "
+        "for a specific instrument and data level. Example usage: "
+        '"python run_processing swe l1a".'
+    )
+
+    instrument_help = f"The instrument to process. Acceptable values are: {instruments}"
+    level_help = (
+        f"The data level to process. Acceptable values are: {processing_levels}"
+    )
+
+    parser = argparse.ArgumentParser(description=description)
```

Comment:
This looks good. An example of the command that I was planning on using for the batch job will be:

['CodiceHi', {'l2': ['2023-10-31', '2023-06-12'], 'l3': ['2023-06-01']}, 2]

where 2 is the version number. 

And then I was planning on passing in the following environment:

                environment={
                    "OUTPUT_PATH": data_bucket.bucket_name,
                    "SECRET_NAME": db_secret_name,
                },

Reply:
@greglucas as I understand it the algorithms will still need to query a database table, for example the universal spin table or a calibration table.

I was thinking that we had to append the s3 bucket information to the lambda url in order to use the upload/download api's. But I could be mistaken. Is that not correct?

Reply:
> @greglucas as I understand it the algorithms will still need to query a database table, for example the universal spin table or a calibration table.

I think we should avoid leaking anything database related between the processing and infrastructure. Can we make new endpoints for this instead?
`/calibration_lookup`, `/spin_table_lookup`, ... whatever else we need

> I was thinking that we had to append the s3 bucket information to the lambda url in order to use the upload/download api's. But I could be mistaken. Is that not correct?

I think the Lambda should take care of this for us? So a user should only need to know the URL, but the details of where we put that in the backend shouldn't matter to them.

`dev.imap-processing.com/upload` should map to a different bucket than `prod.imap-processing.com/upload`, but a user shouldn't need to update both the url and bucket name. I'm hoping we can handle that for them.

Reply:
@greglucas we could do as you suggested for the ENTRYPOINT, but to avoid duplicates I think we should do

imap-processing --instrument codice-hi --level 2 --dates [list of dates]

Reply:
We should talk about this at your processing trigger discussion meeting. If I am reprocessing an entire mission, do I want to put 5 x 365 dates into the command line (there may even be a limit for how long the input terminal can be), versus a start / end date? Could the processing code handle the duplicates? Yet another option is to only allow one date in. So we'd spin up 5 x 365 separate processing jobs all with one date input...